components/userProfile/profileImg.tsx

```diff
@@ -0,0 +1,50 @@
+import Image from "next/image";
+
+import getSpotifyData from "../../lib/getSpotifyData";
+
+const ProfileImg = async ({ username }: { username: string | undefined }) => {
+  let img;
```

Comment:
Don't use let — try something like if (!username throw an error), and then move the rest outside of the if statement. That'll let you have a const and also give an error if the component isn't used properly.